### PR TITLE
Add release notes for NuttX-10.0.0

### PR DESCRIPTION
**Pull request by btashton**
## Summary
This contains the release notes for the 10.0.0 release.

### CODE REVIEW

**Comment by btashton** [comment]
This is currently fully in sync with the draft in the wiki which is also up-to-date as of now.  There is one outstanding ticket that I know of that needs to be addressed before we merge this in and cut an RC.

**Comment by Ouss4** [comment]
Did you manually convert it to markdown?

We need to do the same to the [template](https://cwiki.apache.org/confluence/display/NUTTX/Release+Notes+Template+-+NuttX+X.Y) and then be able to just copy paste from Confluence.

**Comment by btashton** [comment]
> Did you manually convert it to markdown?
> 
> We need to do the same to the [template](https://cwiki.apache.org/confluence/display/NUTTX/Release+Notes+Template+-+NuttX+X.Y) and then be able to just copy paste from Confluence.

I agree it was annoying and I would say I did a marginal/okay job.  Basically I exported the confluence page as HTML then used turndown https://domchristie.github.io/turndown/ tweaked it a bit and ran it through rewrap to get the line lengths under control.

It would be better if we just did it in markdown in the first place.  I think the issues there might be I dont know how well Confluence lets multiple people edit at the same place.  There might be a better solution for that.  Something we can think about going forward.

**Comment by Ouss4** [comment]
> It would be better if we just did it in markdown in the first place. I think the issues there might be I dont know how well Confluence lets multiple people edit at markdown the same time. There might be a better solution for that. Something we can think about going forward.

Are you aware of any limitations in Confluence about that?  For a reference, podling reports are all in Markdown.  When submitted, it's just one file that's being edited by multiple people at the same time.
Anyway, we'll talk about this (and hopefully the ReleaseNote file too) after we get 10.0 out.

File: ReleaseNotes
Comment (acassis) [inline ReleaseNotes:27238]: Chang = Change

File: ReleaseNotes
Comment (acassis) [inline ReleaseNotes:27437]: configuration

File: ReleaseNotes
Comment (acassis) [inline ReleaseNotes:27516]: host

BTW, this patch is for simulator, not for ESP32, it is at wrong section

File: ReleaseNotes
Comment (acassis) [inline ReleaseNotes:27791]: initialization

File: ReleaseNotes
Comment (acassis) [inline ReleaseNotes:27794]: is enabled

File: ReleaseNotes
Comment (acassis) [inline ReleaseNotes:27802]: compensation

File: ReleaseNotes
Comment (acassis) [inline ReleaseNotes:27806]: driver

File: ReleaseNotes
Comment (acassis) [inline ReleaseNotes:27810]: selection

File: ReleaseNotes
Comment (acassis) [inline ReleaseNotes:27813]: earlier

File: ReleaseNotes
Comment (acassis) [inline ReleaseNotes:27820]: required

File: ReleaseNotes
Comment (acassis) [inline ReleaseNotes:27851]: incorrect

File: ReleaseNotes
Comment (acassis) [inline ReleaseNotes:27888]: erase/write

File: ReleaseNotes
Comment (acassis) [inline ReleaseNotes:27932]: gauge

File: ReleaseNotes
Comment (acassis) [inline ReleaseNotes:28002]: Ensure

File: ReleaseNotes
Comment (acassis) [inline ReleaseNotes:28044]: received

**Comment by btashton** [comment]
Thanks @acassis I made all your corrections.  Could someone go ahead and merged this @xiaoxiang781216 @Ouss4.  It is only a text file change so I don't think we need to wait for CI.  I'll also open the PR for the 10.0 branch

**Review by xiaoxiang781216 — APPROVED** [review]
LGTM.

**Pull request merged**
